Bugfix/acna 3561

dthampy opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

## Description

Checks for node major version greater than 22 and displays a warning message when a non supported node version is detected

## Related Issue
[725](https://github.com/adobe/aio-cli/issues/725) 

<!--- This project only accepts pull requests related to open issues -->
<!--- If suggesting a new feature or change, please discuss it in an issue first -->
<!--- If fixing a bug, there should be an issue describing it with steps to reproduce -->
<!--- Please link to the issue here: -->

## Motivation and Contex


## How Has This Been Tested?

<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->

## Screenshots (if appropriate):
<img width="888" alt="Screenshot 2025-04-11 at 7 33 58 PM" src="https://github.com/user-attachments/assets/8d50a3b9-8a4a-4368-b941-cf56171879e5" />

## Types of changes

<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

pru55e11 reviewed (dismissed):
LGTM

pru55e11 commented:
@purplecabbage what are your thoughts on placing this check here? Also, do we want a more generic error message that maybe links to the docs about supported version so this file doesn't become a dependency that needs to get updated every year when we add support for the next version?

dthampy commented:
Would love to get some feedback on the verbiage used here. I couldn't find devdocs that point to supported node versions, hence added the link to nvm documentation.
The reason(s) why the check was placed there
(i) By adding the check in src/index.js, we can make sure it runs at the earliest possible stage of CLI execution before any commands are processed
(ii) Also I thought that by placing it in the run method allows the execution to continue while creating a non-intrusive warning to the users

purplecabbage commented:
This is good, I would avoid using the term Adobe I/O anywhere new as we have sort of abandoned it ( even though it IS the aio cli ... )
It might be easiest to just cut it off with "... which is not officially supported." or "Warning: Node version ${nodeVersion} not supported."

purplecabbage commented:
@dthampy regarding your question about how best not to create future work
This might be an opportunity to be more specific with our supported node engines in package.json

```
"engines": {
  "node": "^18 || ^21 || ^22"
},
```
Then instead of looking just for >22 you could do:

```
const nodeVersion = process.version
if (!semver.satisfies(nodeVersion, config.pjson.engines.node)) {
  // output warning
}
```

dthampy commented:
@pru55e11 @purplecabbage 
Simplified the warning to just show the relevant text. 
Also now using the Config object takes care of reading and parsing the package.json file. Updated the unit tests as well

<img width="882" alt="Screenshot 2025-04-15 at 4 32 17 PM" src="https://github.com/user-attachments/assets/158692b3-b062-4869-8a7f-17dc458e12bc" />

pru55e11 reviewed (approved):
Awesome! Thanks @dthampy

purplecabbage commented on src/index.js:
```diff
@@ -25,6 +27,12 @@ AIOCommand.run = async (argv, opts) => {
 
+  // Check Node.js version
+  const nodeVersion = process.version
+  if (!semver.satisfies(nodeVersion, config.pjson.engines.node)) {
```
Is it too geeky to output this?
`Supported versions are ${config.pjson.engines.node}.`
Then we would just have to update package.json and not look at this again :)

dthampy replied:
I thought about that. But wasn't sure if its ok to show the versions from package json. Made the change though
<img width="920" alt="Screenshot 2025-04-15 at 6 58 24 PM" src="https://github.com/user-attachments/assets/60f0b552-6354-4a19-b8d7-81f683c5b5b4" />
